### DISCUSSION TITLE
manipulate the scene (add, move, rotate, remove objects)

### DISCUSSION THREAD

**Discussion by damavand1** [isl-org/Open3D#6706] [General]
hi

this is my code
i want manipulate the scene, add object move them, remove and somethings like this
how can i do ?

the problem is when i use draw_geometries code flow not continue

another my request is when i press some button manipulate the scene (add, move, rotate, remove )

`
import open3d as o3d
from Open3D_CustomMethods import *

i = 0
def main():
    modelA ="D:/data/3d_model.obj"
    modelB ="D:/data/270.obj"

    modelA_mesh = o3d.io.read_triangle_mesh(modelA)
    modelB_mesh = o3d.io.read_triangle_mesh(modelB)

    modelA_mesh.compute_vertex_normals()
    modelB_mesh.compute_vertex_normals()
   
    o3d.visualization.draw_geometries([modelA_mesh])

    # here i want move draw position (100,0,100) 
    # and add a new mesh to scene 
    o3d.visualization.draw_geometries([modelB_mesh])

    #remove modelA from scend

    #here move modelB to (0,0,0)
    #add modelA to scene

main()`

**Comment by damavand1** [comment]
https://stackoverflow.com/questions/62912397/open3d-visualizing-multiple-point-clouds-as-a-video-animation

**Comment by EwingKang** [comment]
Just providing some pointers,
When using c++ of  the code, I can create a [`visualizer::O3DVisualizer`](https://github.com/isl-org/Open3D/blob/main/cpp/open3d/visualization/visualizer/O3DVisualizer.h) class object, which has `AddGeometry()`, `RemoveGeometry()`, `UpdateGeometry()`, etc.
The O3DVisualizer is the default underlying tool for visualizing application [defined here](https://github.com/isl-org/Open3D/blob/main/cpp/open3d/visualization/utility/DrawGeometry.h).
